catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/adapter/RVAdapter.java

```diff
@@ -107,8 +110,13 @@ protected void onCheckBoxClick(int position) {
 
 			selectionManager.setSelectionTo(!currentState, position);
 			notifyItemChanged(position);
-		} else {
-			selectionManager.toggleSelection(position);
+		}  else {
+			if (!isMergeSelection || (isMergeSelection && getSelectedItems().size() < numberMergeSelections)) {
+				selectionManager.toggleSelection(position);
+			} else {
+				selectionManager.setSelectionTo(false, position);
```

Comment:
the default behaviour is now hidden in an if in an if. please flatten this structure, and rethink the boolean flag. since theres now already two of those defining the multiselection behaviour this is getting messy and hard to read.
id suggest a intdef with "MULTI", "SINGLE", "PAIRS" with multi being the default, and instead of setting the boolean flags handing in what behaviour is wanted.